### ISSUE TITLE
Linux impl.

### ISSUE THREAD

**Issue by daidedou** [riccardomarin/FARM#4]
Hi, 

I'm trying to run the FARM code on linux. Until now I was able to do so, but I can't find the right implementation for the gradient of ||CA-B||. I found this code: https://github.com/pitbullil/PFM/blob/master/pfm/norm_21_gradient.cpp, and tried to modify it to have different size as input but I can't make it work (I keep obtaining segfaults after). I verified and it seems to come from this part, since when I use 50 as basis size for M and N (and use the original norm_21_gradient.cpp) it works well. I'm not really good in cpp so I might miss something.
Do you have the version of this cpp file for FARM? Cause right now there's only the mex file for windows.

**Comment by riccardomarin** [comment]
Hi @daidedou, and thank you for your post.
Unfortunately, I have never run FARM on Linux, and I am not an expert in cpp either. However the code looks quite easy to reimplment - one option is to rewrite in Matlab code. The cpp implementation (that I confirm comes from Partial Functional Maps) was mainly due to efficiency reasons, but it was also quite a time ago, and probably the computers can manage it easily in Matlab.

We can try to ping @lcosmo or @erodola to see if they have any idea of quicker edits to that file to manage different sizes for M and N.

**Comment by daidedou** [comment]
Thanks for your answer. Let's wait then. I've put a gist with the modification I tried here, if it could be of any help: 
https://gist.github.com/daidedou/af05aa1065952c44e5208aef14b48c0e

**Comment by daidedou** [comment]
Hi, just to let you know that I solved the issue. I made it working and I don't know exactly what happened (I tried a lot of things and finally got it to work), but my guess is that there was a file on which I did a wrong modification for the basis numbers and made the program crash. Now it's working well. If I find some time, I will make a pull request to explain the steps in order to make it work on linux. 

Thank you anyway, great work!
